be/src/vec/olap/block_reader.cpp

```diff
@@ -168,6 +177,49 @@ Status BlockReader::init(const ReaderParams& read_params) {
     return Status::OK();
 }
 
+Status BlockReader::next_block_with_aggregation(Block* block, MemPool* mem_pool,
+                                                ObjectPool* agg_pool, bool* eof) {
+    return _block_aggregator ? _agg_next_block(block, mem_pool, agg_pool, eof)
```

Comment:
If agg is push down, we'd better to let `_next_block_func` be `BlockReader::_agg_next_block` instead of make this condition judgment each time

Reply:
Great suggestion, I've changed it.